src/Acuminator/Acuminator.Analyzers/StaticAnalysis/BqlParameterMismatch/Helpers/ParametersCounter.cs

```diff
@@ -52,24 +53,46 @@ public ParametersCounter(PXContext pxContext)
 			}
 
 			/// <summary>
-			/// Count parameters in type symbol. Return <c>false</c> if the diagnostic should be stopped
+			/// Count parameters in type symbol retrieved for <see cref="Microsoft.CodeAnalysis.CSharp.Syntax.GenericNameSyntax"/> syntax node. 
+			/// Return <c>false</c> if the diagnostic should be stopped
 			/// </summary>
 			/// <param name="typeSymbol">The type symbol.</param>
 			/// <param name="cancellationToken">(Optional) The cancellation token.</param>
 			/// <returns/>
-			public bool CountParametersInTypeSymbol(ITypeSymbol typeSymbol, CancellationToken cancellationToken = default)
+			public bool CountParametersInTypeSymbolForGenericNode(ITypeSymbol typeSymbol, CancellationToken cancellationToken = default)
 			{
 				if (!IsCountingValid || typeSymbol == null || IsCancelled(cancellationToken))
 					return false;
 				
 				PXCodeType? codeType = typeSymbol.GetCodeTypeFromGenericName();
+				return CountParametersInTypeSymbolBasedOnCodeType(typeSymbol, codeType, cancellationToken);
+			}
+
+			/// <summary>
+			/// Count parameters in type symbol retrieved for <see cref="Microsoft.CodeAnalysis.CSharp.Syntax.IdentifierNameSyntax"/> syntax node. 
+			/// Return <c>false</c> if the diagnostic should be stopped
+			/// </summary>
+			/// <param name="typeSymbol">The type symbol.</param>
+			/// <param name="cancellationToken">(Optional) The cancellation token.</param>
+			/// <returns/>
+			public bool CountParametersInTypeSymbolForIdentifierNode(ITypeSymbol typeSymbol, CancellationToken cancellationToken = default)
+			{
+				if (!IsCountingValid || typeSymbol == null || IsCancelled(cancellationToken))
+					return false;
+
+				PXCodeType? codeType = typeSymbol.GetColoringTypeFromIdentifier();
```

Comment:
Feels strange to see `GetColoringTypeFromIdentifier` in `Acuminator.Analyzers`. Maybe we should rename it to smth like "GetCodeTypeFromIdentifier"?

Reply:
Yes, it was created for coloring but was conveniently reused for BQL analysis. It's a good idea to rename it. But I suggest something more related to BQL like
GetBqlPartTypeFromIdentifier

Reply:
No, we can't make it related only to BQL because there are Action and Graph in enum.
I'll use GetCodeTypeFromIdentifier then

Reply:
Done